runtime/src/main/java/org/corfudb/runtime/clients/ClientResponseHandler.java

```diff
@@ -84,11 +85,18 @@ public boolean handle(ResponseMsg response, ChannelHandlerContext ctx) {
                     // For errors, we always want to completeExceptionally
                     errorHandlerMap.get(errorCase).handle(response, ctx, router);
                 } catch (Throwable e) {
+                    log.warn("Server threw exception for {} with request_id: {}",
+                            response.getPayload().getPayloadCase(),
+                            response.getHeader().getRequestId(),
+                            e);
                     router.completeExceptionally(requestId, e);
                 }
             }
         } else if (handlerMap.containsKey(payloadCase)) {
             try {
+                log.debug("Received {} from the server - {}",
+                        response.getPayload().getPayloadCase(),
+                        TextFormat.shortDebugString(response));
```

Comment:
Note, if the whole message is logged, then we probably don't need to explicitly add the payload case. We can also guard the logging statements where `shortDebugString` is used.

Reply:
Added, thanks!